dyno-core/src/main/java/com/netflix/dyno/connectionpool/impl/lb/HostSelectionWithFallback.java

```diff
@@ -337,13 +340,27 @@ public void initWithHosts(Map<Host, HostConnectionPool<CL>> hPools) {
 		
 		Set<String> remoteDCs = new HashSet<String>();
 
+		/* Initialize the hashtag with the first host (if hashtag is defined) */
+		this.hashtag = hPools.keySet().iterator().next().getHashtag();
+		
 		for (Host host : hPools.keySet()) {
 			String dc = host.getRack();
 			if (localRack != null && !localRack.isEmpty() && dc != null && !dc.isEmpty() && !localRack.equals(dc)) {
 				remoteDCs.add(dc);
 			}
+			/**
+			 * Checking for defined hashtags if all of them are the same.
+			 * If not we need to throw an exception.
+			 */
+			if(this.hashtag != null) {
+			   String hashtagNew = host.getHashtag();
+			   if (!this.hashtag.equals(hashtagNew)){
+		               throw new RuntimeException("Hashtags are different across hosts");
+			   }
+			   this.hashtag = hashtagNew;
+			}
```

Comment:
what happens when the first hashtag is null and subsequent are not? 
What you really want is a "set" data structure and make sure at the end the cardinality is 1.

Reply:
There is going to be a change in the code. But to explain this case, If the first hashtag is null, and the subsequent host hash a defined hashtag a `RuntimeException` will be thrown.